File: コードレビュー/05.マスター/プログラム作成問題：マスター編/20200302_138666938/src/namebattler/Fire.java
package namebattler;

public class Fire extends Magic{
	// =======================
	// フィールド変数
	// =======================
	//ダメージ量
	private int damage;

	// =======================
	// コンストラクタ
	// =======================
	//コンストラクタ
	public Fire() {
		super();
		//名前をセット
		this.name = "ファイアー";
		//使用MPをセット
		this.cost = 10;
	}

	// =======================
	// protected メソッド
	// =======================
	//個別効果
	@Override
	protected void Magieffect(Player target) {
		//ダメージ量を設定(10～30)
		this.damage = this.ran.nextInt(21)+10;
		//ターゲットにダメージ
		System.out.print(target.GetName() + "に" + this.damage + "のダメージ！");
		target.Damage(this.damage);
	}
}

File: コードレビュー/07.Unityアプリ/20200701_USSBB1M39/Assets/Script/Fire.cs
using System;
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class Fire : Magic
{

	// =======================
	// フィールド変数
	// =======================
	// 魔法の名前
	protected string name = "ファイヤー";
	// 消費MP
	protected int usemp = 10;

	// =======================
	// コンストラクタ
	// =======================
	public Fire()
	{

	}

	// =======================
	// Getter / Setter
	// =======================

	// =======================
	// protected メソッド
	// =======================

	// =======================
	// private メソッド
	// =======================

	// =======================
	// public メソッド
	// =======================
	/**
	 * 対称プレーヤーに炎撃を加える
	 * @param activePlayer {@inheritDoc}
	 * @param passivePlayer {@inheritDoc}
	 */
	public override void effect(Player activePlayer, Player passivePlayer)
	{

		// MPが足りている場合の処理

		int damage = UnityEngine.Random.Range(10, 30);

		activePlayer.UseMP(this.usemp);

		LogText.AddLog(string.Format("{0} の {1}！", activePlayer.GetName(), this.name));
		LogText.AddLog($"{passivePlayer.GetName()} に {damage} のダメージ！");

		passivePlayer.Damage(damage);
	}
}